nifi-extension-bundles/nifi-amqp-bundle/nifi-amqp-processors/src/main/java/org/apache/nifi/amqp/processors/AbstractAMQPProcessor.java

```diff
@@ -303,7 +311,8 @@ protected Connection createConnection(ProcessContext context, ExecutorService ex
         final ConnectionFactory cf = new ConnectionFactory();
         cf.setUsername(context.getProperty(USER).evaluateAttributeExpressions().getValue());
         cf.setPassword(context.getProperty(PASSWORD).getValue());
-
+        cf.setMaxInboundMessageBodySize(Integer.parseInt(context.getProperty(MAX_INBOUND_MESSAGE_BODY_SIZE).evaluateAttributeExpressions().getValue()));
+        
```

Comment:
```suggestion
        cf.setMaxInboundMessageBodySize(context.getProperty(MAX_INBOUND_MESSAGE_BODY_SIZE).evaluateAttributeExpressions().asDataSize(DataUnit.B).intValue());

```

Reply:
MaxInboundMessageBodySize is an integer and we need to validate that the input is not out of range.

My suggestion is to use this:
StandardValidators.POSITIVE_INTEGER_VALIDATOR

Reply:
Then you can do something like:

````java
.addValidator(StandardValidators.createDataSizeBoundsValidator(1, Integer.MAX_VALUE))
````

Using the data size approach makes it more user friendly as the user can just provide something like '1 MB', etc.